Firestore/Source/Util/FSTDispatchQueue.h

```diff
@@ -74,6 +74,13 @@ typedef NS_ENUM(NSInteger, FSTTimerID) {
  */
 - (void)verifyIsCurrentQueue;
 
+/**
+ * Declares that a dispatch queue operation is already in progress. To be used only when called
+ * back by some other API directly onto our queue. This allows us to safely dispatch directly onto
+ * the worker queue without destroying the invariants this class helps us maintain.
+ */
+- (void)enterCheckedOperation:(void (^)(void))block;
```

Comment:
FYI- I like "Operation" but so far this code uses "Callback" (whereas web uses "Operation" and Android uses "Task").  This difference is annoying (it pained me to introduce "FSTDelayedCallback", etc.) and I'd be a fan of unifying more on "Operation" or similar. I'm fine with keeping this as enterCheckedOperation, but perhaps add a TODO to unify terminology?